projects/QuotesApplication/app/src/main/java/com/quotesapplication/MainActivity.java

```diff
@@ -0,0 +1,98 @@
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+        createNotificationChannel();
+
+        //OnClick function for immediate quote notification
+        btNotification = (Button)findViewById(R.id.quotesButtonNow);
+        btNotification.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String quote = quotesArray[(int)(Math.random() * 60)];
+                NotificationCompat.Builder builder = new NotificationCompat.Builder(view.getContext(), CHANNEL_ID)
+                        .setSmallIcon(R.drawable.ic_verified)
+                        .setContentTitle("Smart Quotes")
+                        .setContentText(quote)
+                        .setPriority(NotificationCompat.PRIORITY_HIGH)
+                        .setStyle(new NotificationCompat.BigTextStyle().bigText(quote));
+
+                NotificationManager notificationManager = (NotificationManager)getSystemService(Context.NOTIFICATION_SERVICE);
+
+                int notifyId = 1;
+                notificationManager.notify(notifyId, builder.build());
+            }
+        });
+
+
+        //OnClick function for timing quote notifications
+        btTimingNotification = (Button)findViewById(R.id.timingQuotesButton);
+        btTimingNotification.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent qIntent = new Intent (view.getContext(), QuotesIntentService.class);
+                startService(qIntent);
+            }
+        });
+    }
+
+    /**********************************************************************
+        From some API version we need to create a notification channel
+        and throw exception if there is any problem with the file
+    **********************************************************************/
+    private void createNotificationChannel(){
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            CharSequence name = "Ex3_channel";
+            int importance = NotificationManager.IMPORTANCE_HIGH;
+
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
+            NotificationManager notificationManager = getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    /**********************************************************************
+        Create an array that contains a quotes that read from a txt file
+     **********************************************************************/
+    private void createQuotesArray() throws IOException {
+        quotesArray = new String[60];
+        InputStream file = this.getResources().openRawResource(R.raw.quotes);
+        BufferedReader br = new BufferedReader(new InputStreamReader(file));
+        for (int i = 0; i < 60; i++) {
```

Comment:
Just read until the EOF, no need to hardly code the number of lines in the file..